src/OnboardingSPA/pages/Steps/DesignColors/index.js

```diff
@@ -1,19 +1,97 @@
+import { getGlobalStyles } from '../../../utils/api/themes';
+import { store as nfdOnboardingStore } from '../../../store';
+import { LivePreview } from '../../../components/LivePreview';
 import CommonLayout from '../../../components/Layouts/Common';
 import { VIEW_DESIGN_COLORS } from '../../../../constants';
-import { store as nfdOnboardingStore } from '../../../store';
+import { useGlobalStylesOutput } from '../../../utils/global-styles/use-global-styles-output';
 
 const StepDesignColors = () => {
-	const { setDrawerActiveView, setIsDrawerOpened, setIsSidebarOpened } =
-		useDispatch( nfdOnboardingStore );
-	useEffect( () => {
-		setIsSidebarOpened( false );
-		setIsDrawerOpened( true );
-		setDrawerActiveView( VIEW_DESIGN_COLORS );
-	}, [] );
-	return <CommonLayout></CommonLayout>;
+	const location = useLocation();
+	const [isLoaded, setIsLoaded] = useState(false);
+	const [pattern, setPattern] = useState();
+
+	const isLargeViewport = useViewportMatch('medium');
+	const {
+		currentStep,
+		currentData,
+		storedPreviewSettings,
+	} = useSelect((select) => {
+		return {
+			currentStep: select(nfdOnboardingStore).getStepFromPath(
+				location.pathname
+			),
+			currentData:
+				select(nfdOnboardingStore).getCurrentOnboardingData(),
+			storedPreviewSettings:
+				select(nfdOnboardingStore).getPreviewSettings(),
+		};
+	}, []);
+
+	const {
+		setDrawerActiveView,
+		setIsDrawerOpened,
+		setIsSidebarOpened,
+		setIsDrawerSuppressed,
+		updatePreviewSettings,
+	} = useDispatch(nfdOnboardingStore);
+
+	useEffect(() => {
+		if (isLargeViewport) {
+			setIsDrawerOpened(true);
+		}
+		setIsSidebarOpened(false);
+		setIsDrawerSuppressed(false);
+		setDrawerActiveView(VIEW_DESIGN_COLORS);
+	}, []);
+
+	const getStylesAndPatterns = async () => {
+		const pattern = await getPatterns(currentStep.patternId, true);
+		const globalStyles = await getGlobalStyles();
+		let selectedGlobalStyle;
+		if (currentData.data.theme.variation) {
+			selectedGlobalStyle = globalStyles.body.filter(
+				(globalStyle) =>
+					globalStyle.title === currentData.data.theme.variation
+			)[0];
+		} else {
+			selectedGlobalStyle = globalStyles.body[0];
+		}
+		setPattern(pattern?.body);
+		setIsLoaded(true);
+	};
+
+	
+	useEffect(() => {
+		if (!isLoaded) getStylesAndPatterns();
+	}, [isLoaded]);
+
+	return (
+		<CommonLayout className="theme-colors-preview">
+			<div className="theme-colors-preview__title-bar">
+				<div className="theme-colors-preview__title-bar__browser">
+					<span className="theme-colors-preview__title-bar__browser__dot"></span>
+					<span className="theme-colors-preview__title-bar__browser__dot"></span>
+					<span className="theme-colors-preview__title-bar__browser__dot"></span>
+				</div>
+			</div>
+			<div className="theme-colors-preview__live-preview-container">
+				{pattern && (
+					<LivePreview
+						blockGrammer={pattern}
+						styling={'custom'}
+						viewportWidth={1300}
+					/>
+				)}
+			</div>
+		</CommonLayout>
+	);
```

Comment:
Let's enclose this entire JSX inside the DesignStateHandler component that handles the installation of the theme required for the preview. https://github.com/newfold-labs/wp-module-onboarding/blob/release/v1.0.0/src/OnboardingSPA/components/StateHandlers/Design/index.js